src/core/load-fragment.c

```diff
@@ -5684,21 +5684,27 @@ int config_parse_cgroup_socket_bind(
         r = extract_first_word(&rvalue, &word, ":", 0);
         if (r == -ENOMEM)
                 return log_oom();
+        if (r <= 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Unable to parse %s assignment, ignoring: %s", lvalue, rvalue);
+                return 0;
+        }
 
-        if (rvalue)
-                address_family = word;
-
-        if (address_family) {
-                if (streq(address_family, "IPv4"))
+        if (rvalue) {
+                if (streq(word, "IPv4"))
                         af = AF_INET;
-                else if (streq(address_family, "IPv6"))
+                else if (streq(word, "IPv6"))
                         af = AF_INET6;
                 else
                         return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
```

Comment:
Please use `log_syntax()`, and it is better to return `0`.
The same pattern appears after `parse_ip_port_range()` below. Could you also update it?